proto/docs/arbitration.md

```diff
@@ -1,74 +1,125 @@
-## Arbitration
-
-This server implementation of P4Runtime supports master-slave arbitration. In
-P4Runtime each client needs to open a bi-directional stream connection to the
-server using the `StreamChannel` RPC. The client should advertise its election
-id right away using a `MasterArbitrationUpdate` message. The process through
-which election ids are allocated to clients is out-of-scope of P4Runtime and of
-this document. The client with the highest election id is referred to as the
+## Controller Roles and Mastership Arbitration
+
+P4Runtime supports multiple controller roles to partition the control of the forwarding
+plane and P4 objects. For each role, P4Runtime supports master-arbitration across
+replicated instances of that controller (to protect against multi-master scenarios). At
+any given time, only one of the controller instances (for a given role) is considered a master.
+
+### Controller (or P4Runtime *client*) Role
+A given role is identified by `Role.id`.
+The `Role.config` describes the role configuration, i.e. what operations, P4 entities,
+behaviors, etc. are in the scope of a given role. If config is not set (default case),
+it implies that all P4 objects and control behaviors are in scope for that role
+(i.e. full pipeline access).
+
+The Role IDs are defined offline in agreement across the entire control plane.
```

Comment:
maybe we should decide on "ID" or "id" and use it consistently in non-code sections?

Reply:
+1